src/consumerIncentives/saga.test.ts

```diff
@@ -161,16 +172,46 @@ describe('fetchAvailableRewardsSaga', () => {
 
   it('skips fetching rewards for an unverified user for supercharge v2', async () => {
     await expectSaga(fetchAvailableRewardsSaga, fetchAvailableRewards())
+      .withState(createMockStore({}).getState())
       .provide([
         [select(phoneNumberVerifiedSelector), false],
         [select(walletAddressSelector), userAddress],
       ])
-      .not.call(fetchWithTimeout)
+      .not.call.fn(fetchWithTimeout)
+      .run()
+  })
+
+  it('skips fetching rewards in absence of tokens with sufficient balance', async () => {
+    await expectSaga(fetchAvailableRewardsSaga, fetchAvailableRewards())
+      .withState(
```

Comment:
any reason why you prefer to mock the state rather than mock the outcome of the tokens selector in the `provide` block? i guess there's not so much difference but it seems like mocking the selector outcome is easier / easier to read in this case?

Reply:
after introducing this new check, I had to alter other tests to pass, given the new check.

I found it simpler to provide a default state to other tests in this suite instead of explicitly mocking the selector, because that way I don't need to partially recreate slice of state for the selector mock, just can reuse what is already there.

then for this particular case I just altered a part of the state (being consistent in relying on mocked state).

if it is harder to read and reason about, I'm happy to mock a selector instead of state for this particular case. 

@kathaypacific what do you think about other tests in the suite? would it make sense to alter them to mock the selector instead of state as well?

Reply:
i don't have strong feelings on this - i just noted it because i thought about it. so up to your preference, since we haven't been terribly consistent with this.

i guess there are tradeoffs with each approach. mocking the state is more cumbersome and it's harder to see which values are important for the tests (especially relying on the default state by passing in no arguments to the mock store). i prefer using providers in saga tests because it is easier to see which values are changing. you could argue that you get more robust test coverage by mocking the state, because it also exercises the selectors, so that might be a reason why you want to leave it like this.

in terms of verboseness, i think both are similar (you could add a `[select(tokensWithTokenBalanceSelector, networkConfig.defaultNetworkId]),  [mockCeloTokenBalance]],` instead of the mock store)

Reply:
thanks for the suggestion! indeed, it's not that verbose.

I will refactor it back to mock selectors instead of store to keep the existing pattern of using selectors in this particular suite. it will also keep the suite on the "atomic" side of the atomicity-broadness tradeoff. otherwise, we actually had to get rid of other selector mocks in favor of the mocked state to be consistent at least across this one suite 😅 

thanks for highlighting this!